### ISSUE TITLE
No aXe rule to flag multiple accessible h1 headings

### ISSUE THREAD

**Issue by zelliott** [dequelabs/axe-core#1544]
**Expectation:** aXe should have a rule to flag multiple accessible h1 headings.  By accessible, I mean accessible to screen-readers. That is, only one h1 heading should be accessible to a screen reader at any single point in time.

**Actual:** No such rule exists.

**Motivation:** There is some discussion of this on https://github.com/dequelabs/axe-core/issues/220. The conclusion seems to be that there are reasonable use cases of having multiple h1 headings on a single page. I agree with this. My follow-up question to that is: are there reasonable use cases of having multiple h1 headings **accessible to screen readers at the same time** on a single page? This might be confusing, so here are some examples:

Example of bad use case of multiple h1 headings:

Suppose I have a page with two h1 headings that are both accessible by screen readers at the same time. This would violate this rule.

```
<!-- Would violate this rule -->
<body>
  <h1>Page title</h1>
  <h1>Section title</h1>
</body>
```

Example of good use case of multiple h1 headings:

Suppose I have a page with two h1 headings: one is at the top of the page and contains the page title, and the other is within a dialog that overlays on top of the page when it appears. This would be a valid use case of multiple h1 headings, as only one is accessible to a screen reader at any single point in time.

```
<!-- Would not violate this rule -->
<body>
  <h1>Page title</h1>
  <some-dialog>
    <h1>Section title</h1>
  </some-dialog>
</body>
```

Thoughts on this potential aXe rule?

**Comment by zelliott** [comment]
Thanks for the response @WilcoFiers.

I'm curious, do you have an example of a blog with multiple articles on a single page? I know of blogs that initially have a single article, and then upon scrolling past the initial article, load additional articles below. Are these the blogs you are thinking of that are candidates for having multiple h1 headings?

Regardless, I understand why we might be hesitant to always recommend such a rule.  I like the idea of raising this for review instead. Would this mean adding a rule under the tag "review-item"?

**Comment by zelliott** [comment]
Test

**Comment by zelliott** [comment]
@WilcoFiers  Bump on my questions above.

**Comment by dylanb** [comment]
Disallowing multiple H1 headings is not a WCAG requirement per se. This would at best be a best practice and might even have to be a needs review best practice.

**Comment by zelliott** [comment]
Sounds good. What's the process for creating a best-practice / needs-review rule?

**Comment by zelliott** [comment]
Ping on this issue.

**Comment by WilcoFiers** [comment]
This issue has gone pretty stale. We should not have best practice rules that only return things for review. We want the needs review to be meaningful. For this particular rule, it is very much open to individual site owners whether or not multiple headings fits with their content. Often it doesn't, sometimes it does. I don't think there should be a rule in core. The best solution for anyone who wants such a rule is to add a custom rule. You can do so using `axe.configure()`. It should be fairly straightforward to write.
